peer/pendingheap/heap.go

```diff
@@ -166,6 +169,26 @@ func (ph *pendingHeap) pushPeer(ps *peerScore) {
 	heap.Push(ph, ps)
 }
 
+// pushPeerRandom must be called in the context of a lock.
+//
+// It inserts a peer randomly into the heap among equal peer scores. This is
+// expected to be called only by Add.
+//
+// This ensures that batches of peer updates are inserted randomly throughout
+// the heap, preventing hearding behavior that may be observed during batch
+// deployments.
+func (ph *pendingHeap) pushPeerRandom(ps *peerScore) {
+	// We choose a random number between [next-len(peers), next], after
+	// incrementing ph.next
+	ph.next++
+
+	// get a random number within the range of len(peers)
+	random := ph.Rand.Float32() * float32(len(ph.peers))
```

Comment:
int/float conversion is about as expensive as sqrt and almost always avoidable. Consider `Rand.Intn(len(ph.peers))`